### PR TITLE
fixed public report url issue for functional GUI

### PR DESCRIPTION
**Pull request by kothulemayur**
For functional GUI, Summary tab and video details tab shows no session data error and also Public report URl is working when test execution is completed, So that added condition for only functional GUI test report link i.e public test report will generate after test execution.

### CODE REVIEW

File: src/main/java/com/blazemeter/api/explorer/Master.java [hunk 1]
```diff
@@ -51,10 +59,31 @@ public String getPublicReport() throws IOException, InterruptedException {
...
-                extractPublicToken(response.getJSONObject("result")), getId());
+
+        String reportUrl = String.format("/app/?public-token=%s#/masters/%s/summary", extractPublicToken(response.getJSONObject("result")), getId());
+
+        // generated public url for functional test GUI
+        if (testType != null && testType.equals(FUNCTIONAL_GUI_TEST)) {
+            reportUrl = String.format("/app/?public-token=%s#/accounts/-1/workspaces/-1/projects/-1/masters/%s/cross-browser-summary",
+                    extractPublicToken(response.getJSONObject("result")), getId());
+        }
+        return utils.getAddress() + reportUrl;
```
Comment (artem-fedorov) [inline src/main/java/com/blazemeter/api/explorer/Master.java]: you can change this code in the following format:
```
// generated public url for functional test GUI
        if (testType != null && testType.equals(FUNCTIONAL_GUI_TEST)) {
            return String.format("%s/app/?public-token=%s#/accounts/-1/workspaces/-1/projects/-1/masters/%s/cross-browser-summary",
                    utils.getAddress(), extractPublicToken(response.getJSONObject("result")), getId());
        }
        return String.format("%s/app/?public-token=%s#/masters/%s/summary", utils.getAddress(), extractPublicToken(response.getJSONObject("result")), getId());


```

File: src/main/java/com/blazemeter/api/explorer/Master.java [hunk 2]
```diff
@@ -36,10 +36,18 @@
  */
 public class Master extends BZAObject {
 
+    private String testType;
+
+    private final String FUNCTIONAL_GUI_TEST = "functionalGui";
```
Comment (artem-fedorov) [inline src/main/java/com/blazemeter/api/explorer/Master.java]: i suggest to move this constant to Test or Base test class

**Review by artem-fedorov — COMMENTED** [review]
also update Readme.md with new API calls

**Comment by kothulemayur** [comment]
> also update Readme.md with new API calls

The API's  are  already present in README.md file

**Review by artem-fedorov — APPROVED** [review]

**Pull request merged**